src/olympia/addons/forms.py

```diff
@@ -53,12 +53,27 @@ def clean_addon_name(name, instance=None, addon_type=None):
     if not addon_type:
         addon_type = instance.type
 
-    id = reverse_name_lookup(name, addon_type)
+    matches = reverse_name_lookup(name, addon_type)
+
+    if matches is not None:
+        addon_id = matches.keys()[0]
+        locales = matches.values()[0]
+
+        if addon_id and (not instance or addon_id != instance.id):
+            # If we get an id and either there's no instance
+            # or the instance.id != id.
+            if id and (not instance or id != instance.id):
+                if isinstance(name, dict):
+                    # Return locale specific message
+                    msg = _(
+                        'This name is already in use for {}. '
+                        'Please choose another.'.format(
+                            ', '.join(locales)))
```

Comment:
Also I find the message confusing. Maybe change it to make it clear that it's in use by another add-on ? I feel with the added information about the locales it's no longer clear that the problem is because of other add-ons.